scripts/dts/dtlib.py

```diff
@@ -1354,6 +1347,32 @@ class Property:
 
       See the to_*() methods for converting the value to other types.
 
+    type:
+      The type of the property, inferred from the syntax used in the
+      assignment. This is one of the following constants (with example
+      assignments):
+
+        Assignment         | Property.type
+        -------------------+------------------------
+        foo;               | dtlib.TYPE_EMPTY
+        foo = []           | dtlib.TYPE_BYTES
+        foo = [01 02]      | dtlib.TYPE_BYTES
+        foo = /bits/ 8 <1> | dtlib.TYPE_BYTES
+        foo = <1>          | dtlib.TYPE_NUM
+        foo = <>           | dtlib.TYPE_NUMS
+        foo = <1 2 3>      | dtlib.TYPE_NUMS
+        foo = <1 2>, <3>   | dtlib.TYPE_NUMS
```

Comment:
I wonder if this case should be TYPE_NUMS_MATRIX.

Reply:
For context, I added it because we have stuff like `reg/interrupts = <1 2>, <3 4>` all over, and it felt a bit mean to forbid that.

Maybe it could be tweaked later if we ever want to forbid many `<>` and just allow a single one though.

Reply:
Its not that we should forbit many `<>`.  Is that I think we should introduce `type: uint32-matrix` for `reg`, `ranges`, and `interrupts`.

Reply:
How would it work differently from `type: array`?

Reply:
Is it just whether you allow many `<>` or not?